From f166ef32f15b079122cb21fc549eab8421292365 Mon Sep 17 00:00:00 2001
From: Konstantin Podsvirov <konstantin@podsvirov.pro>
Date: Thu, 27 Aug 2015 14:38:17 +0300
Subject: [PATCH 2/3] Forced include headers for Qt5 moc on files without
 extension

Note from moc documentation:
-f[<file>] - Force the generation of an #include statement in the output.
             This is the default for header files whose extension starts
             with H or h.
---
 src/applications/osgearth_package_qt/CMakeLists.txt | 11 ++++++-----
 src/applications/osgearth_qt/CMakeLists.txt         | 11 ++++++-----
 src/applications/osgearth_qt_simple/CMakeLists.txt  | 11 ++++++-----
 src/applications/osgearth_qt_windows/CMakeLists.txt | 11 ++++++-----
 src/osgEarthQt/CMakeLists.txt                       | 14 +++++++-------
 5 files changed, 31 insertions(+), 27 deletions(-)

diff --git a/src/applications/osgearth_package_qt/CMakeLists.txt b/src/applications/osgearth_package_qt/CMakeLists.txt
index ba6ceb0..5b9ccb1 100644
--- a/src/applications/osgearth_package_qt/CMakeLists.txt
+++ b/src/applications/osgearth_package_qt/CMakeLists.txt
@@ -18,11 +18,12 @@ IF(Qt5Widgets_FOUND)
     QT5_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
     QT5_WRAP_UI( UI_HDRS ${UI_FILES} )
     QT5_WRAP_CPP( UI_SRCS ${UI_HDRS} )
-    IF(Qt5Widgets_VERSION VERSION_LESS 5.2.0)
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} OPTIONS "-f" )
-    ELSE()
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} )
-    ENDIF()
+    SET(MOC_SRCS)
+    FOREACH( MOC_HDR ${MOC_HDRS} )
+        GET_FILENAME_COMPONENT( MOC_HDR_ABS ${MOC_HDR} ABSOLUTE )
+        QT5_WRAP_CPP( MOC_SRC ${MOC_HDR} OPTIONS "-f${MOC_HDR_ABS}" )
+        LIST( APPEND MOC_SRCS ${MOC_SRC} )
+    ENDFOREACH()
 ELSE()
     INCLUDE( ${QT_USE_FILE} )
     QT4_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
diff --git a/src/applications/osgearth_qt/CMakeLists.txt b/src/applications/osgearth_qt/CMakeLists.txt
index 0e6cf54..871948d 100644
--- a/src/applications/osgearth_qt/CMakeLists.txt
+++ b/src/applications/osgearth_qt/CMakeLists.txt
@@ -11,11 +11,12 @@ set(LIB_QT_RCS
 
 IF(Qt5Widgets_FOUND)
     QT5_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
-    IF(Qt5Widgets_VERSION VERSION_LESS 5.2.0)
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} OPTIONS "-f" )
-    ELSE()
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} )
-    ENDIF()
+    SET(MOC_SRCS)
+    FOREACH( MOC_HDR ${MOC_HDRS} )
+        GET_FILENAME_COMPONENT( MOC_HDR_ABS ${MOC_HDR} ABSOLUTE )
+        QT5_WRAP_CPP( MOC_SRC ${MOC_HDR} OPTIONS "-f${MOC_HDR_ABS}" )
+        LIST( APPEND MOC_SRCS ${MOC_SRC} )
+    ENDFOREACH()
 ELSE()
     INCLUDE( ${QT_USE_FILE} )
     QT4_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
diff --git a/src/applications/osgearth_qt_simple/CMakeLists.txt b/src/applications/osgearth_qt_simple/CMakeLists.txt
index 3b8249e..17f55aa 100644
--- a/src/applications/osgearth_qt_simple/CMakeLists.txt
+++ b/src/applications/osgearth_qt_simple/CMakeLists.txt
@@ -8,11 +8,12 @@ SET(MOC_HDRS
 )
 
 IF(Qt5Widgets_FOUND)
-    IF(Qt5Widgets_VERSION VERSION_LESS 5.2.0)
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} OPTIONS "-f" )
-    ELSE()
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} )
-    ENDIF()
+    SET(MOC_SRCS)
+    FOREACH( MOC_HDR ${MOC_HDRS} )
+        GET_FILENAME_COMPONENT( MOC_HDR_ABS ${MOC_HDR} ABSOLUTE )
+        QT5_WRAP_CPP( MOC_SRC ${MOC_HDR} OPTIONS "-f${MOC_HDR_ABS}" )
+        LIST( APPEND MOC_SRCS ${MOC_SRC} )
+    ENDFOREACH()
 ENDIF()
 
 IF(QT4_FOUND)
diff --git a/src/applications/osgearth_qt_windows/CMakeLists.txt b/src/applications/osgearth_qt_windows/CMakeLists.txt
index b466d36..597f6e6 100644
--- a/src/applications/osgearth_qt_windows/CMakeLists.txt
+++ b/src/applications/osgearth_qt_windows/CMakeLists.txt
@@ -4,11 +4,12 @@ SET(MOC_HDRS
 )
 
 IF(Qt5Widgets_FOUND)
-    IF(Qt5Widgets_VERSION VERSION_LESS 5.2.0)
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} OPTIONS "-f" )
-    ELSE()
-        QT5_WRAP_CPP( MOC_SRCS ${MOC_HDRS} )
-    ENDIF()
+    SET(MOC_SRCS)
+    FOREACH( MOC_HDR ${MOC_HDRS} )
+        GET_FILENAME_COMPONENT( MOC_HDR_ABS ${MOC_HDR} ABSOLUTE )
+        QT5_WRAP_CPP( MOC_SRC ${MOC_HDR} OPTIONS "-f${MOC_HDR_ABS}" )
+        LIST( APPEND MOC_SRCS ${MOC_SRC} )
+    ENDFOREACH()
 ELSE()
     INCLUDE( ${QT_USE_FILE} )
     QT4_WRAP_CPP( MOC_SRCS ${MOC_HDRS} OPTIONS "-f" )
diff --git a/src/osgEarthQt/CMakeLists.txt b/src/osgEarthQt/CMakeLists.txt
index dd451bb..29e2138 100644
--- a/src/osgEarthQt/CMakeLists.txt
+++ b/src/osgEarthQt/CMakeLists.txt
@@ -37,13 +37,13 @@ set(LIB_QT_UIS
 IF(Qt5Widgets_FOUND)
     QT5_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
     QT5_WRAP_UI( LIB_UI_HDRS ${LIB_QT_UIS} )
-    IF(Qt5Widgets_VERSION VERSION_LESS 5.2.0)
-        QT5_WRAP_CPP( LIB_UI_SRCS ${LIB_UI_HDRS} OPTIONS "-f" )
-        QT5_WRAP_CPP( LIB_MOC_SRCS ${LIB_MOC_HDRS} OPTIONS "-f" )
-    ELSE()
-        QT5_WRAP_CPP( LIB_UI_SRCS ${LIB_UI_HDRS} )
-        QT5_WRAP_CPP( LIB_MOC_SRCS ${LIB_MOC_HDRS} )
-    ENDIF()
+    QT5_WRAP_CPP( LIB_UI_SRCS ${LIB_UI_HDRS} )
+    SET(LIB_MOC_SRCS)
+    FOREACH( LIB_MOC_HDR ${LIB_MOC_HDRS} )
+        GET_FILENAME_COMPONENT( LIB_MOC_HDR_ABS ${LIB_MOC_HDR} ABSOLUTE )
+        QT5_WRAP_CPP( LIB_MOC_SRC ${LIB_MOC_HDR} OPTIONS "-f${LIB_MOC_HDR_ABS}" )
+        LIST( APPEND LIB_MOC_SRCS ${LIB_MOC_SRC} )
+    ENDFOREACH()
 ELSE()
     INCLUDE( ${QT_USE_FILE} )
     QT4_ADD_RESOURCES( LIB_RC_SRCS ${LIB_QT_RCS} )
-- 
2.6.3

